assets/js/components/TourTooltips.js

```diff
@@ -96,6 +106,33 @@ export default function TourTooltips( { steps, tourID } ) {
 		dismissTour( tourID );
 	};
 
+	const trackAllTourEvents = ( { index, action, lifecycle, size, status, type } ) => {
+		// The index is 0-based, but step numbers are 1-based.
+		const stepNumber = index + 1;
+
+		if ( type === EVENTS.TOOLTIP && lifecycle === LIFECYCLE.TOOLTIP ) {
+			trackEvent( gaEventCategory, GA_ACTIONS.VIEW, stepNumber );
+		} else if ( status === STATUS.FINISHED && type === EVENTS.TOUR_END && size === stepNumber ) {
+			// Here we need to additionally check the size === stepNumber because
+			// it is the only way to differentiate the status/event combination
+			// from an identical combination that happens immediately after completion
+			// on index `0` to avoid duplicate measurement.
+			trackEvent( gaEventCategory, GA_ACTIONS.COMPLETE, stepNumber );
+		}
+
+		if ( lifecycle !== LIFECYCLE.COMPLETE || status === STATUS.FINISHED ) {
+			return;
+		}
+
+		if ( action === ACTIONS.CLOSE ) {
+			trackEvent( gaEventCategory, GA_ACTIONS.DISMISS, stepNumber );
+		} else if ( action === ACTIONS.PREV ) {
+			trackEvent( gaEventCategory, GA_ACTIONS.PREV, stepNumber );
+		} else if ( action === ACTIONS.NEXT ) {
+			trackEvent( gaEventCategory, GA_ACTIONS.NEXT, stepNumber );
+		}
```

Comment:
This is pretty minor, but do these need to be `else if`s really? It can only ever be one.

I tend to prefer `if` statements over `switch/case`s, and this is fine, but I think technically a bunch of `if`s would work too 🤷🏻‍♂️

Reply:
Good point; they obviously don't need to be, but I think it somewhat ended up like that when the logic for each was more complex and it was simplified later. I generally prefer a single dictionary look-up with 1 or no condition in cases like these 😄  That way the mapping doesn't even need to live in the component itself.